sled-agent/src/instance.rs

```diff
@@ -535,8 +541,32 @@ impl InstanceRunner {
                             self.terminate(mark_failed).await;
                         },
                     }
-
                 },
+                // Requests to terminate the instance take priority over any
```

Comment:
Maybe a silly question, but: What happens if this task is already stuck awaiting a response to one of the commands below? IIUC that's what we're seeing in #6911--the `InstanceRunner` loop for some instance is stuck waiting on a Propolis request, so it's not looking at *any* of its message queues.

Reply:
If we are waiting at an await point, notifying the future that is at an await point inside the select should still wake the task and we will go down the newly-notified select branch.

Reply:
Ah, yeah, you're right, this will need to also select over any operation we do in any of these branches and the termination channel firing. I'll fix that.

Reply:
Okay, @gjcolombo, commit  https://github.com/oxidecomputer/omicron/pull/6915/commits/2a331abecc32e8c07db48a8d45e9388c599d3806 changes this so that the termination channel firing will take priority over any _in flight_ request to the instance, so even if that gets stuck, we will still pull the plug immediately.

I wondered if we wanted to add a grace period to allow the in-flight operation to finish, but after talking to @smklein, I don't think that's actually necessary, because `terminate` is only used by the most forceful attempts to stop an instance (the `vmm_unregister` API and killing an instance that was using an expunged disk). A "normal" attempt to stop the instance should go through `instance_put_state` with the `Stopped` state, which goes in the normal request queue.